### PR TITLE
Migrate auto discovery test to wildfly13

### LINKED ISSUE #99: Fix broken auto-discovery tests
Auto discovery tests are broken. The test is reported as successful but it was not executed. 

I would like to migrate it from swarm to something else e.g. wildfly? The other question is do we want to keep this test? It was useful when we didn't integrate with servlet filter now filter has to be added manually.

### PR DESCRIPTION
**Pull request by pavolloffay**
Resolves #99 

Signed-off-by: Pavol Loffay <ploffay@redhat.com>

### CODE REVIEW

**Comment by pavolloffay** [comment]
Test fails on WF14 on:

```
12:58:55,888 ERROR [org.jboss.msc.service.fail] (ServerService Thread Pool -- 8) MSC000001: Failed to start service jboss.deployment.unit."test.war".undertow-deployment: org.jboss.msc.service.StartException in service jboss.deployment.unit."test.war".undertow-deployment: java.lang.RuntimeException: java.lang.NullPointerException
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentService$1.run(UndertowDeploymentService.java:81)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:1985)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1487)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1378)
	at java.lang.Thread.run(Thread.java:748)
	at org.jboss.threads.JBossThread.run(JBossThread.java:485)
Caused by: java.lang.RuntimeException: java.lang.NullPointerException
	at io.undertow.servlet.core.DeploymentManagerImpl.deploy(DeploymentManagerImpl.java:252)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentService.startContext(UndertowDeploymentService.java:96)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentService$1.run(UndertowDeploymentService.java:78)
	... 8 more
Caused by: java.lang.NullPointerException
	at org.wildfly.microprofile.opentracing.smallrye.TracerInitializer.addJaxRsIntegration(TracerInitializer.java:76)
	at org.wildfly.microprofile.opentracing.smallrye.TracerInitializer.contextInitialized(TracerInitializer.java:67)
	at io.undertow.servlet.core.ApplicationListeners.contextInitialized(ApplicationListeners.java:187)
	at io.undertow.servlet.core.DeploymentManagerImpl$1.call(DeploymentManagerImpl.java:216)
	at io.undertow.servlet.core.DeploymentManagerImpl$1.call(DeploymentManagerImpl.java:185)
	at io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:42)
	at io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43)
	at org.wildfly.extension.undertow.security.SecurityContextThreadSetupAction.lambda$create$0(SecurityContextThreadSetupAction.java:105)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1502)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1502)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1502)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1502)
	at io.undertow.servlet.core.DeploymentManagerImpl.deploy(DeploymentManagerImpl.java:250)
	... 10 more

12:58:55,958 ERROR [org.jboss.as.controller.management-operation] (management-handler-thread - 1) WFLYCTL0013: Operation ("add") failed - address: ([("deployment" => "test.war")]) - failure description: {"WFLYCTL0080: Failed services" => {"jboss.deployment.unit.\"test.war\".undertow-deployment" => "java.lang.RuntimeException: java.lang.NullPointerException
    Caused by: java.lang.RuntimeException: java.lang.NullPointerException
    Caused by: java.lang.NullPointerException"}}
12:58:55,963 ERROR [org.jboss.as.server] (management-handler-thread - 1) WFLYSRV0021: Deploy of deployment "test.war" was rolled back with the following failure message: 
{"WFLYCTL0080: Failed services" => {"jboss.deployment.unit.\"test.war\".undertow-deployment" => "java.lang.RuntimeException: java.lang.NullPointerException
    Caused by: java.lang.RuntimeException: java.lang.NullPointerException
    Caused by: java.lang.NullPointerException"}}
```

I tihnk it collides with WF-MicroProfile OpenTracing integration,

**Comment by pavolloffay** [comment]
@jpkrohling could you please review?

File: opentracing-jaxrs2-itest/auto-discovery/pom.xml [hunk 1]
```diff
@@ -12,57 +12,101 @@
...
+      <groupId>org.jboss.arquillian.container</groupId>
+      <artifactId>arquillian-container-spi</artifactId>
       <scope>test</scope>
     </dependency>
     <dependency>
-      <groupId>io.opentracing</groupId>
-      <artifactId>opentracing-util</artifactId>
+      <groupId>org.wildfly.arquillian</groupId>
+      <artifactId>wildfly-arquillian-container-managed</artifactId>
+      <version>2.1.1.Final</version>
```
Comment (jpkrohling) [inline opentracing-jaxrs2-itest/auto-discovery/pom.xml]: For consistency's sake, it probably belongs to a property

**Comment by jpkrohling** [comment]
@pavolloffay wrote:
> I tihnk it collides with WF-MicroProfile OpenTracing integration,

@jstourac, would you be able to migrate this test to the WF-MP OpenTracing test suite, to make sure we aren't doing anything wrong there?

**Pull request merged**